lib/platform/apple.js

```diff
@@ -133,6 +133,7 @@ const invalidBundleIdentifierCharacter = /[^A-Za-z0-9.-]/g
 
 function plist(os, name, version) {
   const isMac = os === 'macos'
+  const cleanedVersion = version.split('-').at(0)
```

Comment:
With this we end up making assumptions about the parts of the semantic version that should be removed, and those assumptions are currently incomplete. The issue would still be present for build metadata (https://semver.org/#spec-item-10), for example.

We're in luck though as Apple has precisely defined the valid format of the `CFBundle(Short)Version` properties: https://developer.apple.com/documentation/bundleresources/information-property-list/cfbundleversion. We can therefore put together a simple RegEx that pulls out only the parts that are allowed and truncates all the bits that aren't, regardless of what those might be.